machine/v1/types_powervsprovider.go

```diff
@@ -0,0 +1,174 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	// PowerVSResourceTypeID enum property to identify an ID type resource reference
+	PowerVSResourceTypeID PowerVSResourceType = "ID"
+	// PowerVSResourceTypeName enum property to identify an Name type resource reference
+	PowerVSResourceTypeName PowerVSResourceType = "Name"
+	// PowerVSResourceTypeRegEx enum property to identify a tags type resource reference
+	PowerVSResourceTypeRegEx PowerVSResourceType = "RegEx"
+)
+
+// PowerVSResourceType enum attribute to identify the type of resource reference
+type PowerVSResourceType string
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// PowerVSMachineProviderConfig is the type that will be embedded in a Machine.Spec.ProviderSpec field
+// for a PowerVS virtual machine. It is used by the PowerVS machine actuator to create a single Machine.
+//
+// Compatibility level 1: Stable within a major release for a minimum of 12 months or 3 minor releases (whichever is longer).
+// +openshift:compatibility-gen:level=1
+// +k8s:openapi-gen=true
+type PowerVSMachineProviderConfig struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// userDataSecret contains a local reference to a secret that contains the
+	// UserData to apply to the instance.
+	// +optional
+	UserDataSecret *PowerVSSecretReference `json:"userDataSecret,omitempty"`
+
+	// credentialsSecret is a reference to the secret with IBM Cloud credentials.
+	// +optional
+	CredentialsSecret *PowerVSSecretReference `json:"credentialsSecret,omitempty"`
+
+	// serviceInstance is the reference to the Power VS service on which the server instance(VM) will be created.
+	// Power VS service is a container for all Power VS instances at a specific geographic region.
+	// serviceInstance can be created via IBM Cloud catalog or CLI.
+	// supported serviceInstance identifier in PowerVSResource are Name and ID and that can be obtained from IBM Cloud UI or IBM Cloud cli.
+	// More detail about Power VS service instance.
+	// https://cloud.ibm.com/docs/power-iaas?topic=power-iaas-creating-power-virtual-server
+	// +kubebuilder:validation:=Required
+	ServiceInstance PowerVSResource `json:"serviceInstance"`
+
+	// image is to identify the rhcos image uploaded to IBM COS bucket which is used to create the instance.
+	// supported image identifier in PowerVSResource are Name and ID and that can be obtained from IBM Cloud UI or IBM Cloud cli.
+	// +kubebuilder:validation:=Required
+	Image PowerVSResource `json:"image"`
+
+	// network is the reference to the Network to use for this instance.
+	// supported network identifier in PowerVSResource are Name, ID and RegEx	 and that can be obtained from IBM Cloud UI or IBM Cloud cli.
+	// +kubebuilder:validation:=Required
+	Network PowerVSResource `json:"network"`
+
+	// keyPairName is the name of the KeyPair to use for SSH.
```

Comment:
I'd like some more explanation as to how this affects the end user
```suggestion
	// keyPairName is the name of the KeyPair to use for SSH.
	// The key pair will be exposed to the instance via the instance metadata service.
	// On boot, the OS will copy the public keypair into the authorized keys for the core user.
```

Reply:
comment by Joel from a slack thread:

Reply:
PowerVS platform is not really in the list https://github.com/coreos/afterburn/blob/main/systemd/afterburn-sshkeys%40.service.in here, @Prashanth684 can we get the support added for PowerVS in that afterburn sshkeys?

Reply:
sshkeys are configured through ignition when provided through the install config. are we saying we want this support in afterburn in case it is not specified through install config ?